File: api/internal/infrastructure/persistence/league.go
package persistence

type League struct {
	Base

	Name     string         `gorm:"not null;uniqueIndex:ux_league_name" json:"name"`
	IsActive bool           `gorm:"default:true" json:"isActive"`
	Members  []LeagueMember `gorm:"foreignKey:LeagueID" json:"members"`
	Games    []Game         `gorm:"foreignKey:LeagueID" json:"games"`
}

File: bouncy-web/src/lib/models/league.ts
import type { Game } from './game';
import type { LeagueMember } from './league-member';

export interface League {
  id: string;
  name: string;
  isActive: boolean;
  members: LeagueMember[];
  games: Game[];
}

export interface LeagueFinancialSummary {
  leagueId: string;
  totalCollected: number;
  totalCharges: number;
  totalAllocated: number;
  totalUnpaid: number;
  totalAvailable: number;
}
